server/memstore.go

```diff
@@ -666,11 +671,19 @@ func (ms *memStore) PurgeEx(subject string, sequence, keep uint64) (purged uint6
 // Purge will remove all messages from this store.
 // Will return the number of purged messages.
 func (ms *memStore) Purge() (uint64, error) {
+	ms.mu.RLock()
+	first := ms.state.LastSeq + 1
+	ms.mu.RUnlock()
+	return ms.purge(first)
+}
+
+func (ms *memStore) purge(fseq uint64) (uint64, error) {
 	ms.mu.Lock()
 	purged := uint64(len(ms.msgs))
 	cb := ms.scb
 	bytes := int64(ms.state.Bytes)
-	ms.state.FirstSeq = ms.state.LastSeq + 1
+	ms.state.FirstSeq = fseq
```

Comment:
You need to formalize this in case first is not > last.

So stream of 1-100, purge to 50..

Reply:
And in that case you can't just wipe the ms.msgs. Or you could error in memory store for partial purge.

Reply:
For now I'll go with erroring on partial purge, as I think the only alternative is to iterate through the entire `ms.msgs` map and check for lower sequences.